go/runtime/host/sgx/common/common.go

```diff
@@ -155,18 +157,27 @@ func AttestationWorker(
 	t := time.NewTicker(interval)
 	defer t.Stop()
 
+	var retryTicker *backoff.Ticker
+
 	logger = logger.With("runtime_id", hp.Runtime.ID())
 
 	// Get the event emitter.
 	eventEmitter, _ := hp.Runtime.(host.RuntimeEventEmitter)
 
 	for {
+		var retryCh <-chan time.Time
+		if retryTicker != nil {
```

Comment:
I would keep this code together, e.g.
```go
	var (
		retryTicker *backoff.Ticker
		retryCh     <-chan time.Time
	)

	for {
		capabilityTEE, err := updateCapabilityFunc(context.Background(), hp)
		if err != nil {
			// Setup a retry ticker so we retry attestation faster than the configured interval.
			if retryTicker == nil {
				expBackoff := backoff.NewExponentialBackOff(
					backoff.WithMaxElapsedTime(0), // Never stop.
				)
				retryTicker = backoff.NewTicker(expBackoff)
				retryCh = retryTicker.C
			}

			continue
		}

		// Clear retry ticker after successful attestation.
		if retryTicker != nil {
			retryTicker.Stop()
			retryTicker = nil
			retryCh = nil
		}
	}
```

Reply:
It is not taken from committee node, at least not on purpose. Where did the select go?

Reply:
Also if we do it like this we need to always reset one more thing.

Reply:
>  Where did the select go?

Removed it for brevity. 

> Also if we do it like this we need to always reset one more thing.

Yes, but we have one `if` less and less scattered code for retry.